cli/2024-06-04-aws-irsa-support.md

```diff
@@ -0,0 +1,326 @@
+# Radius AWS Workload Identity Support
+
+* **Status**: In Review
+* **Author**: Nithya Subramanian (@nithyatsu)
+
+## Overview
+
+A software workload such as a container-based application, service or script needs an identity to authenticate, access, and communicate with services that are distributed across different platforms and/or cloud providers. Radius uses AWS IAM (Identity and Access Management) credentials to deploy and access AWS resources today:
+ 
+```
+rad credential register aws --access-key-id <access-key-id> --secret-access-key <secret-access-key>
+```
+ 
+These IAM credentials, including the access key should be rotated regularly to reduce the chance of unauthorized access. A more secure option than using these credentials is to use IRSA.
+ 
+IRSA (IAM Roles for Service Accounts) is used for authenticating applications running within Kubernetes pods. When we use IRSA, we associate an IAM role directly with a Kubernetes service-account, allowing the pods associated with the service-account to assume that role. IRSA does not require us to configure access key and  secret. It relies identity tokens (OIDC tokens) to authenticate with AWS services. These tokens have a default expiration (configurable). However, they automatically refresh when needed, transparent to the user.
+
+The goal of the scenario is to enable infrastructure operators to configure IRSA support for the AWS provider in Radius to deploy and manage AWS resources.
+
+## Terms and definitions
+
+
```

Comment:
Any terms we should define here?

Reply:
I think we should define some of the vocabulary used in the doc to describe IAM and it's features. eg: role ARN (I'm sure there are others)

Reply:
e.g. ODIC

Reply:
added them.